### PR TITLE
Add unit tests for com.hankcs.hanlp.algorithm.EditDistance

### PR DESCRIPTION
**Pull request by ThomasPerkins1123**
## 注意事项

* 这次修改没有引入第三方类库。
* 也没有修改JDK版本号
* 所有文本都是UTF-8编码
* 代码风格一致
* [ x] 我在此括号内输入x打钩，代表上述事项确认完毕。

Hi,

I've analysed your code base and noticed that `com.hankcs.hanlp.algorithm.EditDistance` in the `hanlp` module is not fully tested.

I've written some tests that cover this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

### CODE REVIEW

File: src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    @Test
+    public void testComputeList() {
+        Assert.assertEquals(32L,
+            EditDistance.compute(
+                new ArrayList<SynonymItem>(Arrays.asList(
+                    new CommonSynonymDictionary.SynonymItem(
+                        new Synonym("", 64L), null, '='))),
+                new ArrayList<SynonymItem>(Arrays.asList(
+                    new CommonSynonymDictionary.SynonymItem(
+                        new Synonym("", 32L), null, '=')))));
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java]: I would do this instead, as it is easier to read and you can do it without `Arrays.asList()`

        ArrayList<SynonymItem> synonymItems1 = new ArrayList<SynonymItem>();
        synonymItems1.add(new SynonymItem(new Synonym("", 32L), null, '='));

        ArrayList<SynonymItem> synonymItems2 = new ArrayList<SynonymItem>();
        synonymItems2.add(new SynonymItem(new Synonym("", 64L), null, '='));
        Assert.assertEquals(32L,
            EditDistance.compute(synonymItems2, synonymItems1));

**Review by Braavos6 — APPROVED** [review]

File: src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java (hunk 1 above)
Comment (paulkennethkent) [reply src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java]: I agree, inlining this is a bit too difficult to read.

**Review by paulkennethkent — CHANGES_REQUESTED** [review]

**Review by JohnLBergqvist — APPROVED** [review]

File: src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package com.hankcs.hanlp.algorithm;
+
+import com.hankcs.hanlp.corpus.synonym.Synonym;
+import com.hankcs.hanlp.dictionary.common.CommonSynonymDictionary.SynonymItem;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.ArrayList;
+
+
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java]: Remove blank line

File: src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    public void computeString() {
+        Assert.assertEquals(2,
+            EditDistance.compute("foo", "oof"));
+    }
+
+    @Test
+    public void computeList() {
+        ArrayList<SynonymItem> synonymItems1 = new ArrayList<SynonymItem>();
+        synonymItems1.add(new SynonymItem(new Synonym("", 32L), null, '='));
+        ArrayList<SynonymItem> synonymItems2 = new ArrayList<SynonymItem>();
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java:27]: Add a blank line to separate the two arraylists

File: src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import com.hankcs.hanlp.dictionary.common.CommonSynonymDictionary.SynonymItem;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.ArrayList;
+
+
+public class EditDistanceTest {
+
+    @Test
+    public void computeCharArray() {
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java]: Add test to the test method name? I was there in your previous commit

File: src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        Assert.assertEquals(2,
+            EditDistance.compute("foo".toCharArray(), "oof".toCharArray()));
+    }
+
+    @Test
+    public void computeString() {
+        Assert.assertEquals(2,
+            EditDistance.compute("foo", "oof"));
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/algorithm/EditDistanceTest.java]: This can be in one line.. but if you prefer you can keep it like this :)

**Review by Braavos6 — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

**Pull request merged**